### PR TITLE
INTG-3044 course progress feeds

### PR DESCRIPTION
**Pull request by MickStanciu**
## Summary

This PR implements the course progress feeds.
Also changes the way we report remaining items 

## Checklist

- [ ] Is this PR a reasonable size?

- [ ] List deployment sequence with all relevant PRs.
  - ...

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: pkg/api/fixtures/test_limit_101.yaml [hunk 1]
```diff
@@ -16,6 +16,9 @@ export:
     limit: 101
   issue:
     limit: 101
+  course:
+    progress:
+      limit: 1001
```
Comment (sc-gv) [inline pkg/api/fixtures/test_limit_101.yaml:21]: should this be 101?

Comment (MickStanciu) [reply pkg/api/fixtures/test_limit_101.yaml:21]: no, default limit for courses is 1000

**Review by sc-gv — APPROVED** [review]
